sourcecode/hub/app/Http/Controllers/Admin/AdminsController.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Requests\GrantAdminRequest;
+use App\Models\User;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Response;
+
+use function redirect;
+use function response;
+
+final readonly class AdminsController
+{
+    public function index(): Response
+    {
+        $users = User::where('admin', true)->paginate();
+
+        return response()->view('admin.admins.index', [
+            'users' => $users,
+        ]);
+    }
+
+    public function add(GrantAdminRequest $request): RedirectResponse
+    {
+        $email = $request->validated('email');
+
+        $user = User::where('email', $email)->firstOrFail();
+        $user->admin = true;
+        $user->save();
+
+        return redirect()->back();
+    }
+
+    public function remove(User $user): RedirectResponse
```

Comment:
There should always be at least one admin, i.e. the last admin can not be demoted

Reply:
Admins can always be added via artisan, there's no risk of total lockout.